services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java

```diff
@@ -2137,36 +2138,31 @@ public void rollForwardToFutureVersion(String clusterName, String storeName, Str
           "Sending roll forward command to future version {} for store {} to child controllers",
           futureVersionBeforeRollForward,
           storeName);
-      sendAdminMessageAndWaitForConsumed(clusterName, storeName, message);
+      Set<String> failedRegions = new HashSet<>();
+      Map<String, ControllerClient> controllerClients = getVeniceHelixAdmin().getControllerClientMap(clusterName);
+      for (Map.Entry<String, ControllerClient> entry: controllerClients.entrySet()) {
+        ControllerClient controllerClient = entry.getValue();
+        RetryUtils.executeWithMaxAttemptAndExponentialBackoff(() -> {
+          ControllerResponse response = controllerClient.rollForwardToFutureVersion(storeName, regionFilter);
```

Comment:
I dont think controller client will throw exception in the sad path. You need to check for `response.isError()` as you are checking below and throw some exception which can be retried.